tests/test_initialization.py

```
@@ -1264,6 +1264,23 @@ def test_lora_with_bias_incompatible_arguments(self, extra_kwargs):
         with pytest.raises(ValueError, match=msg):
             LoraConfig(target_modules=["linear"], lora_bias=True, **extra_kwargs)
 
+    def test_lora_forbidden_mamba_modules(self):
+        # Ensure LoRA raises an error when applying to forbidden modules
+        # ('out_proj', 'conv1d') in Mamba-based architectures like Falcon-H1.
+        model = AutoModelForCausalLM.from_pretrained("tiiuae/Falcon-H1-0.5B-Instruct")
+
+        peft_config = LoraConfig(
+            task_type="CAUSAL_LM",
+            inference_mode=False,
+            r=8,
+            lora_alpha=16,
+            lora_dropout=0.1,
+            target_modules=["out_proj", "conv1d"],  # Forbidden modules for Mamba-based models
+        )
+
+        # This should raise the error if your patched PEFT is in effect
+        model = get_peft_model(model, peft_config)
```

With the new changes, this should raise an error, right? So let's do:

```python
msg = ...
with pytest.raises(ValueError, match=msg):
    model = get_peft_model(model, peft_config)
```